kube-runtime/src/watcher.rs

```diff
@@ -205,6 +217,14 @@ pub struct Config {
     /// Configures re-list for performance vs. consistency.
     pub list_semantic: ListSemantic,
 
+    /// Maximum number of objects retrieved per list operation resyncs.
+    ///
+    /// This can reduce the memory consumption during resyncs, at the cost of requiring more
+    /// API roundtrips to complete.
+    ///
+    /// Defaults to unlimited (everything is retrieved in one request) if `None`.
```

Comment:
It may be worth setting a limit by default, e.g., 500 (like in the Kubernetes client-go). There are not so many cases when users want to retrieve all the data without pagination (I doubt I know one 😅 ).

Reply:
Aligning with client-go probably makes sense, yeah.

Reply:
Continued in #1249 where this is has been set to 500